tests/c/icmp_ptr.newcg.c

```diff
@@ -0,0 +1,42 @@
+// ignore-if: test $YK_JIT_COMPILER != "yk" -o "$YKB_TRACER" = "swt"
+// Run-time:
+//   env-var: YKD_SERIALISE_COMPILATION=1
+//   env-var: YKD_LOG_JITSTATE=-
+//   stderr:
+//     ...
+//     jitstate: enter-jit-code
+//     p1==p2: 1
+//     p1==p2: 1
+//     jitstate: deoptimise
+//     ...
+//
+
+// Check that pointer comparisons work.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk.h>
+#include <yk_testing.h>
+
+int main(int argc, char **argv) {
+  YkMT *mt = yk_mt_new(NULL);
+  yk_mt_hot_threshold_set(mt, 0);
+  YkLocation loc = yk_location_new();
+
+  int i = 0;
+  int *p1 = &i, *p2 = &i;
+  NOOPT_VAL(loc);
+  NOOPT_VAL(i);
+  NOOPT_VAL(p1);
+  NOOPT_VAL(p2);
+  while (i < 4) {
+    yk_mt_control_point(mt, &loc);
+    fprintf(stderr, "p1==p2: %d\n", p1 == p2);
```

Comment:
Should we add another comparison that returns false?

Reply:
Try 13af6a61066d